### ISSUE TITLE
Auto backup file timestamps not changing

### ISSUE THREAD

**Issue by alexgreendesign** [rqlite/rqlite#1833]
**What version are you running?**
8.26.6

**Are you using Docker or Kubernetes to run your system?**
No

**Are you running a single node or a cluster?**
Cluster

Running a cluster with -auto-backup:

```
{
	"version": 1,
	"type": "s3",
	"interval": "5m",
	"timestamp": true,
	"vacuum": false,
	"sub": {
		...
		"path": "/backups/primary/db_primary.sqlite3.gz"
	}
}
```

And from the status:

```
auto_backups": {
	"last_index": "773",
	"last_upload_duration": "1.791925657s",
	"last_upload_time": "2024-07-12T06:49:35Z",
	"upload_destination": "s3://...db_primary.sqlite3.gz",
	"upload_interval": "5m0s"
},
```

The backups use the initial timestamp, and continue to overwrite the same file (creating versions).

![Screenshot 2024-07-12 at 12 33 47 PM](https://github.com/user-attachments/assets/88a8f1a0-6447-4e74-8fb3-df3ea74518f6)

**Comment by otoolep** [comment]
Thanks for the report -- I'm looking into this now.

**Comment by otoolep** [comment]
Fixed, I'm going to make a new release (v8.26.7) once the PR is merged.

This was a bad bug, there should have been better test coverage (which I have now fixed). Timestamping must not be used that widely yet as I'm surprised this wasn't reported earlier.
